bbq/compiler/desugar.go

```diff
@@ -1044,8 +1047,10 @@ func (d *Desugar) interfaceDelegationMethodCall(
 	}
 
 	invocationTypes := sema.InvocationExpressionTypes{
-		ReturnType:    funcType.ReturnTypeAnnotation.Type,
-		ArgumentTypes: funcType.ParameterTypes(),
+		ReturnType: funcType.ReturnTypeAnnotation.Type,
+		// TODO:
+		ArgumentTypes:  funcType.ParameterTypes(),
```

Comment:
Think this is safe and correct. Here we are simply delegation call, inside the synthetic-wrapper function we created. Something like:
```cadence
fun defaultFunc(a1: T1, a2: T2): R {
    return Interface.defaultFunc(a1, a2)
}
```

Where `defaultFunc` wrapper is created using the same parameter types as `Interface.defaultFunc`. So the argument types to the invocation `Interface.defaultFunc` is same as the parameter type `defaultFunc`/`Interface.defaultFunc`.

Reply:
Also see: https://github.com/onflow/cadence/blob/6bc99e707156ce628141e0192e36cf240cb0c659/bbq/compiler/desugar.go#L846-L855

Reply:
Ah, I see, great! Replaced the TODO with a comment with this explanation, thanks! c07a3a54c814fbb69db1302a2d9371c33714cf38